reference/runners/docker.rst

```diff
@@ -0,0 +1,717 @@
+.. _reference_runners_docker:
+
+Docker runner
+=============
+
+How to use a docker runner
+--------------------------
+
+To run conan inside a docker container you need to define a ``[runner]`` section in your host profile file using the following fields.
+
+- ``type`` **(mandatory)**: define the ruuner we want to use, in that case ``docker``.
+- ``dockerfile`` **(optional, default None)**: define the Dockerfile absolute path in case you want to build a docker image.
+- ``image`` **(optional, default conan-runner-default)**: docker image name you want to download from a docker registry or the name of the builded image in case you define a dockerfile path.
+- ``cache`` **(optional, default clean)**: how docker container uses (or not) the host cache.
+
+    - ``clean``: use an empty cache.
+    - ``copy``: copy the host cache inside the container using the :ref:`conan cache save/restore<reference_commands_cache>` command.
+    - ``shared``: mount the host conan cache as a shared volume.
+
+- ``remove`` **(optional, default false)**: ``true`` or ``false``. Remove the container after running the conan command.
+- ``configfile``  **(optional, default None)**: configuration file absolute path with extra parameters (see **extra configuration** section for more info).
+
+Extra configuration
+-------------------
+
+If you need more controll over the container build and execution, you can define more parameters inside a ``configfile`` yaml.
+
+..  code-block:: yaml
+
+    image: image_name # The image to build or run.
+    build:
+        dockerfile: /dockerfile/path # Dockerfile path.
+        build_context: /build/context/path # Path within the build context to the Dockerfile.
+        build_args: # A dictionary of build arguments
+            foo: bar
+        cacheFrom: # A list of images used for build cache resolution
+            - image_1
+    run:
+        name: container_name # The name for this container.
+        containerEnv: # Environment variables to set inside the container.
+            env_var_1: env_valie
+        containerUser: user_name # Username or UID to run commands as inside the container.
+        privileged: False # Run as privileged
+        capAdd: # Add kernel capabilities.
+            - SYS_ADMIN
+            - MKNOD
+        securityOpt: # A list of string values to customize labels for MLS systems, such as SELinux.
+            - opt_1
+        mount: # A dictionary to configure volumes mounted inside the container.
+            /home/user1/: # The host path or a volume name
+                bind: /mnt/vol2 # The path to mount the volume inside the container
+                mode: rw # rw to mount the volume read/write, or ro to mount it read-only.
+
+Basic example
```

Comment:
It's ok to show basic instructions here, but the best thing would be to make a complete example in the examples section and also with matching code in the [examples2 repo](https://github.com/conan-io/examples2)